### ISSUE TITLE
batch transform Not working local mode.

### ISSUE THREAD

**Issue by toyama0919** [aws/sagemaker-python-sdk#953]
### System Information

* Framework: PyTorch
* Framework Version: 1.1.0
* Python Version:py3
* CPU or GPU: Both
* Python SDK Version: 1.34.2.dev0
* Are you using a custom image: Yes, N/A
* batch transform Not working local mode.

### Describe the problem

batch transform Not working local mode.

this code fixing below, worked.
https://github.com/aws/sagemaker-python-sdk/blob/master/src/sagemaker/local/data.py#L236

```
        with open(file, "rb") as f:
```

why not support binary?

#### input data(jpeg)
```
➜  mnist git:(master) ✗ ls -l evaluation
-rwxr-xr-x   1 hiroshi.toyama  1522739515  593  5 11  2017 img_108.jpg
-rwxr-xr-x@  1 hiroshi.toyama  1522739515  540  5 11  2017 img_122.jpg
-rwxr-xr-x@  1 hiroshi.toyama  1522739515  614  5 11  2017 img_12289.jpg
-rwxr-xr-x@  1 hiroshi.toyama  1522739515  635  5 11  2017 img_127.jpg
-rwxr-xr-x@  1 hiroshi.toyama  1522739515  574  5 11  2017 img_140.jpg
-rwxr-xr-x@  1 hiroshi.toyama  1522739515  615  5 11  2017 img_2.jpg
-rwxr-xr-x@  1 hiroshi.toyama  1522739515  467  5 11  2017 img_22.jpg
-rwxr-xr-x@  1 hiroshi.toyama  1522739515  558  5 11  2017 img_34.jpg
-rwxr-xr-x@  1 hiroshi.toyama  1522739515  562  5 11  2017 img_8.jpg
-rwxr-xr-x@  1 hiroshi.toyama  1522739515  576  5 11  2017 img_96.jpg
```

#### code
```py
    model = PyTorchModel(**parameters)

    transformer = model.transformer(
        1,
        instance_type,
        accept="application/x-tensor",
        output_path="file://{}/output/".format(os.getcwd()),
        # output_path="s3://sagemaker-us-east-1-549338637798/output/",
        strategy="SingleRecord",
        assemble_with=None,
        output_kms_key=None,
        env=None,
        max_concurrent_transforms=None,
        max_payload=None,
        tags=None,
        volume_kms_key=None,
    )
    transformer.transform(
        "file://{}/evaluation/".format(os.getcwd()),
        # "s3://sagemaker-us-east-1-xxxxxxxxxxx/sagemaker/evaluation/",
        data_type='S3Prefix',
        content_type="application/x-image",
        compression_type=None,
        split_type=None,
        job_name=None,
        input_filter=None,
        output_filter=None,
        join_source=None
    )
    transformer.wait()

```

#### error stacktrace
```
algo-1-gg4i0_1  | Building wheels for collected packages: mnist, configobj
algo-1-gg4i0_1  |   Running setup.py bdist_wheel for mnist ... done
algo-1-gg4i0_1  |   Stored in directory: /tmp/pip-ephem-wheel-cache-udfbclfy/wheels/35/24/16/37574d11bf9bde50616c67372a334f94fa8356bc7164af8ca3
algo-1-gg4i0_1  |   Running setup.py bdist_wheel for configobj ... done
algo-1-gg4i0_1  |   Stored in directory: /root/.cache/pip/wheels/f1/e4/16/4981ca97c2d65106b49861e0b35e2660695be7219a2d351ee0
algo-1-gg4i0_1  | Successfully built mnist configobj
algo-1-gg4i0_1  | Installing collected packages: wurlitzer, websocket-client, configobj, everett, netifaces, comet-git-pure, comet-ml, mnist
algo-1-gg4i0_1  | Successfully installed comet-git-pure-0.19.11 comet-ml-2.0.5 configobj-5.0.6 everett-1.0.2 mnist-1.0.0 netifaces-0.10.9 websocket-client-0.56.0 wurlitzer-1.0.3
algo-1-gg4i0_1  | You are using pip version 18.1, however version 19.2.1 is available.
algo-1-gg4i0_1  | You should consider upgrading via the 'pip install --upgrade pip' command.
algo-1-gg4i0_1  | 172.20.0.1 - - [26/Jul/2019:08:33:53 +0000] "GET /ping HTTP/1.1" 200 0 "-" "-"
algo-1-gg4i0_1  | 172.20.0.1 - - [26/Jul/2019:08:33:53 +0000] "GET /execution-parameters HTTP/1.1" 404 2 "-" "-"
Traceback (most recent call last):
  File "./04_batch_predict.py", line 85, in <module>
    main(args)
  File "./04_batch_predict.py", line 76, in main
    join_source=None
  File "/Users/hiroshi.toyama/github/sagemaker-python-sdk/src/sagemaker/transformer.py", line 163, in transform
    join_source,
  File "/Users/hiroshi.toyama/github/sagemaker-python-sdk/src/sagemaker/transformer.py", line 298, in start_new
    data_processing=data_processing,
  File "/Users/hiroshi.toyama/github/sagemaker-python-sdk/src/sagemaker/session.py", line 613, in transform
    self.sagemaker_client.create_transform_job(**transform_request)
  File "/Users/hiroshi.toyama/github/sagemaker-python-sdk/src/sagemaker/local/local_session.py", line 124, in create_transform_job
    transform_job.start(TransformInput, TransformOutput, TransformResources, **kwargs)
  File "/Users/hiroshi.toyama/github/sagemaker-python-sdk/src/sagemaker/local/entities.py", line 164, in start
    self._perform_batch_inference(input_data, output_data, **kwargs)
  File "/Users/hiroshi.toyama/github/sagemaker-python-sdk/src/sagemaker/local/entities.py", line 302, in _perform_batch_inference
    for item in batch_provider.pad(file, max_payload):
  File "/Users/hiroshi.toyama/github/sagemaker-python-sdk/src/sagemaker/local/data.py", line 342, in pad
    for element in self.splitter.split(file):
  File "/Users/hiroshi.toyama/github/sagemaker-python-sdk/src/sagemaker/local/data.py", line 225, in split
    yield f.read()
  File "/Users/hiroshi.toyama/.pyenv/versions/3.7.2/lib/python3.7/codecs.py", line 322, in decode
    (result, consumed) = self._buffer_decode(data, self.errors, final)
UnicodeDecodeError: 'utf-8' codec can't decode byte 0xff in position 0: invalid start byte
```

**Comment by toyama0919** [comment]
```
        with open(file, "rb") as f:
```

if this fixed working container.


```
algo-1-2tchk_1  | Building wheels for collected packages: mnist
algo-1-2tchk_1  |   Running setup.py bdist_wheel for mnist ... done
algo-1-2tchk_1  |   Stored in directory: /tmp/pip-ephem-wheel-cache-epn4qucy/wheels/35/24/16/37574d11bf9bde50616c67372a334f94fa8356bc7164af8ca3
algo-1-2tchk_1  | Successfully built mnist
algo-1-2tchk_1  | Installing collected packages: mnist
algo-1-2tchk_1  |   Found existing installation: mnist 1.0.0
algo-1-2tchk_1  |     Uninstalling mnist-1.0.0:
algo-1-2tchk_1  |       Successfully uninstalled mnist-1.0.0
algo-1-2tchk_1  | Successfully installed mnist-1.0.0
algo-1-2tchk_1  | You are using pip version 18.1, however version 19.2.1 is available.
algo-1-2tchk_1  | You should consider upgrading via the 'pip install --upgrade pip' command.
algo-1-2tchk_1  | 172.20.0.1 - - [26/Jul/2019:08:47:41 +0000] "POST /invocations HTTP/1.1" 200 369 "-" "-"
algo-1-2tchk_1  | 172.20.0.1 - - [26/Jul/2019:08:47:42 +0000] "POST /invocations HTTP/1.1" 200 369 "-" "-"
algo-1-2tchk_1  | 172.20.0.1 - - [26/Jul/2019:08:47:42 +0000] "POST /invocations HTTP/1.1" 200 369 "-" "-"
algo-1-2tchk_1  | 172.20.0.1 - - [26/Jul/2019:08:47:42 +0000] "POST /invocations HTTP/1.1" 200 369 "-" "-"
algo-1-2tchk_1  | 172.20.0.1 - - [26/Jul/2019:08:47:42 +0000] "POST /invocations HTTP/1.1" 200 369 "-" "-"
algo-1-2tchk_1  | 172.20.0.1 - - [26/Jul/2019:08:47:42 +0000] "POST /invocations HTTP/1.1" 200 369 "-" "-"
algo-1-2tchk_1  | 172.20.0.1 - - [26/Jul/2019:08:47:42 +0000] "POST /invocations HTTP/1.1" 200 369 "-" "-"
algo-1-2tchk_1  | 172.20.0.1 - - [26/Jul/2019:08:47:42 +0000] "POST /invocations HTTP/1.1" 200 369 "-" "-"
algo-1-2tchk_1  | 172.20.0.1 - - [26/Jul/2019:08:47:42 +0000] "POST /invocations HTTP/1.1" 200 369 "-" "-"
Gracefully stopping... (press Ctrl+C again to force)
!
```

**Comment by toyama0919** [comment]
image processing example.(jpg => jpg.out)

https://aws.amazon.com/jp/blogs/news/sagemaker-nysummit2018/

**Comment by toyama0919** [comment]
I think make BinarySplitter.(read binary)

**Comment by icywang86rui** [comment]
@toyama0919 
Just trying to clarify what's the issue here.

Does the example in the blogpost work for your out of the box? Did you try the same model with localmode and get the decoding error?

**Comment by jesterhazy** [comment]
fixed by #954
